run.py

```diff
@@ -409,31 +411,80 @@ def run_diffusion_processsing(**args):
                 if stage in args.stages:
                     stage_func()
 
-        dwis = layout.get(subject=subject_label, type='dwi',
-                                                 extensions=["nii.gz", "nii"])
-
-        # print(dwis)
-        # acqs = set(layout.get(target='acquisition', return_type='id',
-        #                       subject=subject_label, type='dwi',
-        #                       extensions=["nii.gz", "nii"]))
-        # print(acqs)
-        # posData = []
-        # negData = []
-        # for acq in acqs:
-        #     pos = "EMPTY"
-        #     neg = "EMPTY"
-        #     dwis = layout.get(subject=subject_label,
-        #                       type='dwi', acquisition=acq,
-        #                       extensions=["nii.gz", "nii"])
-        #     assert len(dwis) <= 2
-        #     for dwi in dwis:
-        #         dwi = dwi.filename
-        #         if "-" in layout.get_metadata(dwi)["PhaseEncodingDirection"]:
-        #             neg = dwi
-        #         else:
-        #             pos = dwi
-        #     posData.append(pos)
-        #     negData.append(neg)
-        #
-        # print(negData)
-        # print(posData)
+        posData = []
+        negData = []
+        PEdir = "None"
+        dwiname = "Diffusion"
+        dirnums = []
+        onerun = False
+
+        numruns = set(layout.get(target='run', return_type='id',
+                                 subject=subject_label, type='dwi',
+                                 extensions=["nii.gz", "nii"]))
+        # accounts for multiple runs, number of directions, and phase encoding directions
+        if not numruns:
+            onerun= True
+            numruns = {'run-01'}
+        if numruns:
+            for session in numruns:
```

Comment:
I think this would make more sense since you are iterating over runs not sessions.